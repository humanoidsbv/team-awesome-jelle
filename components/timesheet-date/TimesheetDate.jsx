import React from 'react';

import PropTypes from 'prop-types';

import { calculateCumulativeDuration } from '../../services/convert-time/convert-time';

import './timesheet-date.scss';

const TimesheetDate = ({ date, startTime, timesheetEntries }) => (
  <React.Fragment>
    <div className="timesheet-date">
      <p className="timesheet-date__date">
        {`
          ${new Date(date).toLocaleDateString('en-NL', { weekday: 'long' })}
          ${new Date(date).toLocaleDateString('nl-NL', { day: 'numeric', month: 'numeric' })}
        `}
      </p>
      <p className="timesheet-date__total-time">
        {calculateCumulativeDuration(startTime, timesheetEntries)}
      </p>
    </div>
  </React.Fragment>
);

TimesheetDate.propTypes = {
  timesheetEntries: PropTypes.arrayOf(PropTypes.shape({
    employer: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    startTime: PropTypes.string.isRequired,
    endTime: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired
  })).isRequired,
  date: PropTypes.string.isRequired,
  startTime: PropTypes.string.isRequired
};

export default TimesheetDate;
