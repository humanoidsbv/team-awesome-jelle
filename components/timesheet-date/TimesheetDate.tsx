import React from 'react';

import './timesheet-date.scss';

interface TimesheetDateProps {
  date: string;
  totalTime: string;
}

const TimesheetDate = ({ date, totalTime } : TimesheetDateProps) => (
  <h2 className="timesheet-date">
    <span className="timesheet-date__date">
      {`
        ${new Date(date).toLocaleDateString('en-NL', { weekday: 'long' })}
        ${new Date(date).toLocaleDateString('nl-NL', { day: 'numeric', month: 'numeric' })}
      `}
    </span>
    <span className="timesheet-date__total-time">
      {totalTime}
    </span>
  </h2>
);

export default TimesheetDate;
