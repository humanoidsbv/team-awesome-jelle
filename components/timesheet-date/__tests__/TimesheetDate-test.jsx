import React from 'react';
import { shallow } from 'enzyme';

import TimesheetDate from '../TimesheetDate.tsx';

const date = '2018-12-13';

it('Renders Date Component correctly', () => {
  const wrapper = shallow(
    <TimesheetDate
      date={date}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
