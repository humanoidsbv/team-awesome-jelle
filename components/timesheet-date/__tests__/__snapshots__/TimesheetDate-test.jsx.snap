// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Date Component correctly 1`] = `
<p
  className="timesheet-wrapper__date"
>
  
      Thursday
      12/13
    
</p>
`;
